common/djangoapps/util/philu_utils.py

```diff
@@ -17,3 +24,26 @@ def extract_utm_params(input_dict):
     ]
 
     return {key: value for key, value in input_dict.items() if key in utm_keys}
+
+
+@deconstructible
+class UploadToPathAndRename(object):
+    """
+    Rename file uploaded by user.
+    """
+
+    def __init__(self, path, field_name):
+        self.sub_path = path
+        self.field_name = field_name
+
+    def __call__(self, instance, filename):
+        file_extension = filename.split('.')[-1] if filename else ''
+
+        name_prefix = 'image' if instance._meta.get_field(self.field_name).__class__ is ImageField else 'file'
+
+        # if pk has been created then use pk otherwise set filename as random string
+        unique_name = instance.pk or uuid4().hex
```

Comment:
why is this an or? I see we've tried to make this generic. But if used in models, wouldnt this always be one case out of these?